src/pages/auth/otp-code/ui/timer-view.tsx

```diff
@@ -0,0 +1,67 @@
+import { styled } from '@shared/ui/theme';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Typography } from '@shared/ui/atoms';
+import { calculateEndTimer, calculateTimeLeft, convertSeconds } from '../lib';
+
+const TimerWrapper = styled.View`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+`
+
+const TimerLabel = styled(Typography)`
+    color: ${({ theme }) => theme.palette.text.secondary};
+`
+
+type Props = {
+    onKeyPress: () => void
+}
+
+const TouchWrapper = styled.TouchableOpacity`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+`
+
+export type TimerInfo = {
+    minutes: number
+    seconds: number
+}
+
+const timerInitial: TimerInfo = { minutes: 3, seconds: 0 }
+
+export const TimerView = ({ onKeyPress }: Props) => {
+    const [isEnded, setIsEnded] = useState(false)
+    const [timeLeft, setTimeLeft] = useState<TimerInfo>(timerInitial)
+    const timeEnd = useRef(calculateEndTimer())
+
+    useEffect(() => {
+        const difference = timeEnd.current - Date.now()
+        const timeout = setTimeout(() => {
+            setTimeLeft(calculateTimeLeft(difference))
+        }, 1000)
+        if (difference <= 0) {
+            clearTimeout(timeout)
+            setIsEnded(true)
+        }
+    }, [timeEnd, timeLeft])
```

Comment:
Не заметил в первый раз
Не уверен, но мне кажется, что стоит очищать таймайт при анмаунте компонента

```suggestion
    useEffect(() => {
        const difference = timeEnd.current - Date.now()
        const timeout = setTimeout(() => {
            setTimeLeft(calculateTimeLeft(difference))
        }, 1000)
        if (difference <= 0) {
            clearTimeout(timeout)
            setIsEnded(true)
        }
        
        return () => clearTimeout(timeout)
    }, [timeEnd, timeLeft])
```

Reply:
Проверь как будет работать